src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+        // TODO: log type (scalar, binary, categorical)
+        // TODO: log any immutable data associated with the market (e.g., endTime, numOutcomes, payoutDenominator, denominationToken address, etc.)
+
+    // this will need to be called manually for each open market if a spender contract is updated
+    function approveSpenders() private returns (bool) {
+        bytes32[5] memory _names = [bytes32("cancelOrder"), bytes32("completeSets"), bytes32("takeOrder"), bytes32("tradingEscapeHatch"), bytes32("claimProceeds")];
+        uint8 i = 0;
+        for (i = 0; i < 5; i++) {
+            denominationToken.approve(controller.lookup(_names[i]), APPROVAL_AMOUNT);
+        }
+        for (i = 0; i < numOutcomes; i++) {
+            shareTokens[i].approve(controller.lookup("takeOrder"), APPROVAL_AMOUNT);
+        }
+        return true;
+    }
+
+    function changeCreator(address _newCreator) public returns (bool) {
+        require(msg.sender == creator);
+        creator = _newCreator;
+        return true;
+    }
+
+    function decreaseMarketCreatorSettlementFeeInAttoethPerEth(uint256 _newFeePerEthInWei) public returns (bool) {
+        require(_newFeePerEthInWei < feePerEthInAttoeth);
+        require(msg.sender == creator);
+        feePerEthInAttoeth = _newFeePerEthInWei;
+        return true;
+    }
+
+    function automatedReport(uint256[] _payoutNumerators) public returns (bool) {
+        // intentionally does not migrate the market as automated report markets won't actually migrate unless a dispute bond has been placed or the automated report doesn't occur
+        require(msg.sender == automatedReporterAddress);
+        require(isInAutomatedReportingPhase());
+        // we have to create the reporting token so the rest of the system works (winning reporting token must exist)
+        getReportingToken(_payoutNumerators);
+        automatedReportReceived = true;
+        tentativeWinningPayoutDistributionHash = derivePayoutDistributionHash(_payoutNumerators);
+        reportingWindow.updateMarketPhase();
+        return true;
+    }
+
+    function disputeAutomatedReport() public returns (bool) {
+        // intentionally does not migrate the market as automated report markets won't actually migrate unless a dispute bond has been placed or the automated report doesn't occur
+        require(isInAutomatedDisputePhase());
+        automatedReporterDisputeBondToken = DisputeBondTokenFactory(controller.lookup("DisputeBondTokenFactory")).createDisputeBondToken(controller, this, msg.sender, AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT, tentativeWinningPayoutDistributionHash);
+        reportingState = ReportingState.LIMITED;
+        fundDisputeBondWithReputation(msg.sender, automatedReporterDisputeBondToken, AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT);
+        reportingWindow.updateMarketPhase();
+        return true;
+    }
+
+    function disputeLimitedReporters() public returns (bool) {
+        migrateThroughAllForks();
```

Comment:
I wonder if it would improve readability to have a modifier for `triggersMigration` that does `migrateThroughAllForks` rather than having it be the first line of a bunch of different functions?

There _may_ be similar readability benefits for things like `onlyInPhase(X)` (we would want an enum for all the possible phases in this case).